cmd/deploy.go

```diff
@@ -59,162 +58,224 @@ type UpRequest struct {
-					resp.StatusCode, body)
-			}
-
-		case daemonStatus:
-			// Get status of the deployment
-			resp, err := deployment.Status(auth)
-			if err != nil {
-				log.WithError(err)
-			}
-
-			defer resp.Body.Close()
-			body, err := ioutil.ReadAll(resp.Body)
-			if err != nil {
-				log.WithError(err)
-			}
-
-			switch resp.StatusCode {
-			case http.StatusOK:
-				fmt.Printf("(Status code %d) %s\n", resp.StatusCode, body)
-			case http.StatusForbidden:
-				fmt.Printf("(Status code %d) Bad auth: %s\n", resp.StatusCode, body)
-			case http.StatusNotFound:
-				fmt.Printf("(Status code %d) Problem with deployment: %s\n", resp.StatusCode, body)
-			case http.StatusPreconditionFailed:
-				fmt.Printf("(Status code %d) Problem with deployment setup: %s\n", resp.StatusCode, body)
-			default:
-				fmt.Printf("(Status code %d) Unknown response from daemon: %s\n",
-					resp.StatusCode, body)
-			}
-
-		case daemonReset:
-			// Remove project from deployment
-			resp, err := deployment.Reset(auth)
-			if err != nil {
-				log.WithError(err)
-			}
-
-			defer resp.Body.Close()
-			body, err := ioutil.ReadAll(resp.Body)
-			if err != nil {
-				log.WithError(err)
-			}
-
-			switch resp.StatusCode {
-			case http.StatusOK:
-				fmt.Printf("(Status code %d) %s\n", resp.StatusCode, body)
-			case http.StatusForbidden:
-				fmt.Printf("(Status code %d) Bad auth: %s\n", resp.StatusCode, body)
-			default:
-				fmt.Printf("(Status code %d) Unknown response from daemon: %s\n",
-					resp.StatusCode, body)
-			}
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.WithError(err)
+		}
 
+		switch resp.StatusCode {
+		case http.StatusOK:
+			fmt.Printf("(Status code %d) %s\n", resp.StatusCode, body)
+		case http.StatusForbidden:
+			fmt.Printf("(Status code %d) Bad auth: %s\n", resp.StatusCode, body)
 		default:
-			fmt.Printf("No such deployment command: %s\n", args[1])
-			os.Exit(1)
+			fmt.Printf("(Status code %d) Unknown response from daemon: %s\n",
+				resp.StatusCode, body)
 		}
 	},
 }
 
+// deployCmd represents the deploy command
+var deployCmd = &cobra.Command{
+	Hidden: true,
+	Long: `Start or stop continuous deployment to the remote VPS instance specified.
+Run 'inertia remote status' beforehand to ensure your daemon is running.
+Requires:
+
+1. A deploy key to be registered for the daemon with your GitHub repository.
+2. A webhook url to registered for the daemon with your GitHub repository.
+
+Run 'inertia [REMOTE] init' to collect these.`,
+}
+
 func init() {
+	// TODO: multiple remotes - loop through and add each one as a
+	// new copy of a command using addRemoteCommand
+	config, err := getProjectConfigFromDisk()
+	if err != nil {
+		return
+	}
+
+	newCmd := &cobra.Command{}
+	newCmd = deployCmd
+	newCmd.AddCommand()
+
+	addRemoteCommand(config.CurrentRemoteName, newCmd)
+}
+
+func addRemoteCommand(remoteName string, cmd *cobra.Command) {
```

Comment:
Super cool!

```bash
▶ ./inertia gcloud --help
Start or stop continuous deployment to the remote VPS instance specified.
Run 'inertia remote status' beforehand to ensure your daemon is running.
Requires:

1. A deploy key to be registered for the daemon with your GitHub repository.
2. A webhook url to registered for the daemon with your GitHub repository.

Run 'inertia [REMOTE] init' to collect these.

Usage:
  inertia gcloud [command]

Available Commands:
  down        Bring project offline on remote
  init        Initialize the VPS for continuous deployment
  reset       Reset the project on your remote
  status      Print the status of deployment on remote
  up          Bring project online on remote

Flags:
  -h, --help   help for gcloud

Use "inertia gcloud [command] --help" for more information about a command.
```

Reply:
Hm, should disable the automatic command sorting so we can order the commands our way

Reply:
9867ed2 ;)